oidc-authenticator/src/main/java/org/xwiki/contrib/oidc/auth/internal/OIDCUserManager.java

```diff
@@ -304,10 +308,129 @@ private Principal updateUser(IDTokenClaimsSet idToken, UserInfo userInfo) throws
+     */
+    protected void removeUserFromXWikiGroup(String xwikiUserName, String groupName, XWikiContext context)
+    {
+        try {
+            BaseClass groupClass = context.getWiki().getGroupClass(context);
+
+            // Get the XWiki document holding the objects comprising the group membership list
+            XWikiDocument groupDoc = context.getWiki().getDocument(groupName, context);
+
+            synchronized (groupDoc) {
+                // Get and remove the specific group membership object for the user
+                BaseObject groupObj =
+                        groupDoc.getXObject(groupClass.getDocumentReference(), XWIKI_GROUP_MEMBERFIELD, xwikiUserName);
+                groupDoc.removeXObject(groupObj);
+
+                // Save modifications
+                context.getWiki().saveDocument(groupDoc, context);
+            }
+        } catch (Exception e) {
+            this.logger.error("Failed to remove a user from a group " + xwikiUserName + " group: " + groupName, e);
+        }
+    }
+
+    protected void addUserToXWikiGroup(String xwikiUserName, String groupName, XWikiContext context)
+    {
+        try {
+            BaseClass groupClass = context.getWiki().getGroupClass(context);
+
+            // Get document representing group
+            XWikiDocument groupDoc = context.getWiki().getDocument("XWiki." + groupName, context);
+
+            this.logger.info("Adding user [{}] to xwiki group [{}]. groupClass: [{}]; groupDoc: [{}]", xwikiUserName, groupName, groupClass, groupDoc);
+
+            synchronized (groupDoc) {
+                // Make extra sure the group cannot contain duplicate (even if this method is not supposed to be called
+                // in this case)
+                List<BaseObject> xobjects = groupDoc.getXObjects(groupClass.getDocumentReference());
+                if (xobjects != null) {
+                    for (BaseObject memberObj : xobjects) {
+                        if (memberObj != null) {
+                            String existingMember = memberObj.getStringValue(XWIKI_GROUP_MEMBERFIELD);
+                            if (existingMember != null && existingMember.equals(xwikiUserName)) {
+                                this.logger.warn("User [{}] already exist in group [{}]", xwikiUserName,
+                                        groupDoc.getDocumentReference());
+                                return;
+                            }
+                        }
+                    }
+                }
+
+                // Add a member object to document
+                BaseObject memberObj = groupDoc.newXObject(groupClass.getDocumentReference(), context);
+                Map<String, String> map = new HashMap<String, String>();
+                map.put(XWIKI_GROUP_MEMBERFIELD, xwikiUserName);
+                groupClass.fromMap(map, memberObj);
+
+                // If the document is new, set its content
+                if (groupDoc.isNew()) {
```

Comment:
This is actually totally useless in recent XWiki version (LDAP authenticator code is very old...).

Reply:
You mean `groupClass.fromMap(map, memberObj);` ?

Reply:
No I mean this if block. Users don't need any content anymore, containing a user object is enough to end up with user displayer.

Reply:
Noted, thanks.